runtime/ms-rest-azure/lib/filters/rpRegistrationFilter.js

```diff
@@ -0,0 +1,160 @@
+                return callback(error, response, response.body);
+              });
+            } else {
+              return callback(err, response, body);
+            }
+          });
+        }
+      }
+      return callback(err, response, body);
+    });
+  };
+};
+
+/**
+ * Validates the error code and message associated with 409 response status code. If it matches to that of 
+ * RP not registered then it returns the name of the RP else returns undefined.
+ * @param {string} body - The response body received after making the original request.
+ * @returns {string} result The name of the RP if condition is satisfied else undefined.
+ */
+exports.checkRPNotRegisteredError = function checkRPNotRegisteredError(body) {
+  let result, responseBody;
+  if (body) {
+    try {
+      responseBody = JSON.parse(body);
+    } catch (err) {
+      //do nothing;
+    }
+    if (responseBody && responseBody.error && responseBody.error.message && 
+    responseBody.error.code && responseBody.error.code === 'MissingSubscriptionRegistration') {
+      let matchRes = responseBody.error.message.match(/'(\w+\.?\w+)'*/ig);
+      if (matchRes && matchRes[1]) {
+        result = matchRes[1].slice(1, matchRes[1].length-1);
+      }
+    }
+  }
+  return result;
+};
+
+/**
+ * Extracts the first part of the URL, just after subscription: 
+ * https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/
+ * @param {string} url - The original request url
+ * @returns {string} urlPrefix The url prefix as explained above.
+ */
+exports.extractSubscriptionUrl = function extractSubscriptionUrl(url) {
+  let result;
+  let matchRes = url.match(/.*\/subscriptions\/[a-f0-9-]+\//ig);
+  if (matchRes && matchRes[0]) { 
+    result = matchRes[0];
+  } else {
+    throw new Error(`Unable to extract subscriptionId from the given url - ${url}.`);
+  }
+  return result;
+};
+
+/**
+ * Registers the given provider.
+ * @param {string} urlPrefix - https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/
+ * @param {string} provider - The provider name to be registered.
+ * @param {object} originalRequest - The original request sent by the user that returned a 409 response
+ * with a message that the provider is not registered.
+ * @param {registrationCallback} callback - The callback that handles the RP registration
+ */
+exports.registerRP = function regiserRP(urlPrefix, provider, originalRequest, callback) {
+  let postUrl = `${urlPrefix}providers/${provider}/register?api-version=2016-02-01`;
+  let getUrl = `${urlPrefix}providers/${provider}?api-version=2016-02-01`;
+  let reqOptions = {
+    headers: originalRequest.headers
+  };
+  reqOptions.headers['x-ms-client-request-id'] = utils.generateUuid();
+  return request.post(postUrl, reqOptions, (err, response) => {
+    if (response.statusCode !== 200) {
+      return callback(new Error(`Autoregistration of ${provider} failed. Please try registering manually.`));

```

Comment:
just checking to see if any other status code is considered a success? also extract to constant if possible